packages/grid/src/HeaderWrapper.tsx

```diff
@@ -0,0 +1,148 @@
+import React, {
+  useState,
+  useRef,
+  useEffect,
+  useCallback,
+  MutableRefObject,
+  MouseEvent,
+} from 'react';
+import { HeaderCellWrapper } from './HeaderCell';
+import { Header, MovingElem } from './styled';
+import { IHeaderWrapper, IColumn } from './interfaces';
+
+
+export const HeaderWrapper = ({
+  fullWidth,
+  translateX,
+  columns,
+  onChangeWidth,
+  onChangeMoving,
+  changeChangingColumns,
+  theme,
+  shouldMovingColumns,
+  shouldChangeColumnsWidth,
+}: IHeaderWrapper) => {
+  const mappedColumns = useRef(columns);
+  const [isMoving, changeIsMoving] = useState(false);
+  const clickX = useRef(0);
+  const movingColumnIndex = useRef<number>();
+  const movingColumnData = useRef<IColumn>();
+  const movingElemRect = useRef<ClientRect>();
+  const headerRef = useRef<HTMLDivElement>() as MutableRefObject<HTMLDivElement>;
+  const emptyColumnIndex = useRef<number>();
+  const [mouseMove, changeMouseMove] = useState(0);
+  const [startCoord, changeStartCoord] = useState({ x: 0, y: 0, height: 0 });
+  const startClickX = useRef(0);
+
+  useEffect(() => {
+    mappedColumns.current = columns;
+    changeStartCoord({ x: 0, y: 0, height: 0 });
+  }, [columns]);
+
+  const handleMouseMove = useCallback(
+    (e) => {
+      const { clientX } = e;
+      const moveMouse = clientX - clickX.current;
+
+      const emptyIdx = emptyColumnIndex?.current || 0;
+
+      if (moveMouse < 0) {
+        if (mappedColumns.current[emptyIdx - 1]) {
```

Comment:
How about add this: `const leftColumnIndex = emptyIdx - 1;`
Logic with `emptyIdx - 1` repeats several times.
Same for `moveMouse > 0` logic - `const rightColumnIndex = emptyIdx + 1;`